src/configuration-factory.ts

```diff
@@ -30,18 +37,24 @@ export function configurationStorageFactory(
   if (forceMemoryOnly) {
     return new MemoryOnlyConfigurationStore();
   } else if (persistentStore) {
-    return new HybridConfigurationStore(new MemoryStore<Flag>(), persistentStore);
+    return new IsolatedHybridConfigurationStore(
+      new MemoryStore<Flag>(),
+      persistentStore,
+      servingStoreUpdateStrategy,
```

Comment:
This "update strategy" tells it how to isolate the stores. The default of "always" makes it behave just like the non-isolated `HybridStore` that updates both stores every call to `setEntries()`.